scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/spel/Typer.scala

```diff
@@ -425,7 +425,7 @@ private[spel] class Typer(
             case result :: Nil =>
               // Limitation: projection on an iterative type makes it loses it's known value,
               // as properly determining it would require evaluating the projection expression for each element (likely working on the AST)
-              projectionResult(iterateType, result)
+              valid(Typed.genericTypeClass[java.util.List[_]](List(result)))
```

Comment:
Please add such a test:
```
    val typedExpr = parse[Any]("'a,b'.split(',').![#this]", ctx).validValue
    val evaluationResult = typedExpr.expression.evaluateSync[Any](ctx)
    val expectedRuntimeClass = typedExpr.returnType.asInstanceOf[SingleTypingResult].objType.klass
    val runtimeResultHasTypeSameAsTyped = expectedRuntimeClass isAssignableFrom evaluationResult.getClass
    runtimeResultHasTypeSameAsTyped shouldBe true
```
and fix it ;)

Reply:
Done

Reply:
I like this check:
```
expectedRuntimeClass isAssignableFrom evaluationResult.getClass
```

more than:
```
parsed.validValue.returnType shouldBe expectedRuntimeType```
```

Because it checks the business requirement not a technical implementation. The requirement is to not blow up the scenario in runtime which implies  that `runtimeObjType` will be the same as the real type returned during evaluation. BTW if think that we could add to every test as a part of `evaluateSync` because it is general rule that should be correct for every case.

WDYT?